apps/twittertrends/twitter_trends.star

```diff
@@ -0,0 +1,263 @@
+"""
+Applet: Twitter Trends
+Summary: Displays top twitter trends
+Description: Displays the top N number of Trending Hashtags on Twitter. Colors of the trends text determine how many tweets it has. White: No Volume Data, Green: Less then 25K, Blue: 25K - 100K, Orange: 100K - 250K, Purple: 250K - 500K, Red: More than 500K.
+Author: Joseph Esposito
+"""
+
+load("cache.star", "cache")
+load("encoding/base64.star", "base64")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("render.star", "render")
+load("schema.star", "schema")
+
+#SCHEMA is erroring on me saying invalid module.
+
+#In my implementation, this is how I connect to the Twitter API using my account Key:Secret Key
+#I don't know if this works with the OAuth2 Schema because I didn't have a way of testing it to my knowledge.
+#In order to get the twitter API keys I applied for an elevated developer account, which was free and easy.
+
+TWITTER_TRENDS_URL = "https://api.twitter.com/1.1/trends/place.json"
+
+TWITTER_ICON = base64.decode("""
+iVBORw0KGgoAAAANSUhEUgAAAAoAAAAICAYAAADA+m62AAAAAXNSR0IArs4c6
+QAAAGJJREFUKFNj5Dj+3OGHpeQBBiQAEmNiZNoPEvr3/58jSJ4RJvjNQpwRpp
+brxMv/yBpBbLAkTAKmGF0hSBxuIroJMD7canzWobgRxEF2PLLJMNPgbsRmKrI
+iuEJ0E9EVgRQCADNVQsMdRdaKAAAAAElFTkSuQmCC
+""")
+
+def main(config):
+    """_summary_
+
+    Args:
+        config (_type_): _description_
+
+    Returns:
+        render: The render for the Tidbyt's display.
+    """
+
+    #Check for cache
+    trends_cached = cache.get("twitter_trends_rate")
+    top_trends = {}
+
+    if trends_cached != None:
+        print("Hit! Displaying cached data.")
+
+        top_trends = json.decode(trends_cached)
+    else:
+        print("Miss! Calling Twitter data.")
+
+        #Commenting out Schema for tests
+
+        #Check for API keys
+
+        if config.get("key", None) != None or config.get("secret", None) != None:
+            api_parameters = {
+                "client_id": "{}:{}".format(config.get("key", None), config.get("secret", None)),
+            }
+        else:
+            return render.Root(
```

Comment:
Any chance you can return a meaningful preview? When we display apps in the apps store, we render the app without any config values, which means all you would see in the app store is "No API Keys!" D:

Reply:
Hi! Should be all set to return a preview of the app now. Let me know if you need anything else!